pkg/clusteragent/admission/controllers/secret/controller.go

```diff
@@ -203,7 +204,7 @@ func (c *Controller) updateSecret(secret *corev1.Secret) error {
 
 	secret = secret.DeepCopy()
 	secret.Data = data
-	_, err = c.clientSet.CoreV1().Secrets(c.config.GetNs()).Update(secret)
+	_, err = c.clientSet.CoreV1().Secrets(c.config.GetNs()).Update(context.TODO(), secret, metav1.UpdateOptions{})
```

Comment:
Should we introduce a timeout on this context here? Basically use `context.WithTimeout(..)` here and in similar spots.